helix-core/src/surround.rs

```diff
@@ -141,15 +144,26 @@ pub fn find_nth_pairs_pos(
 
     let (open, close) = if open == close {
         if Some(open) == text.get_char(pos) {
-            // Cursor is directly on match char. We return no match
-            // because there's no way to know which side of the char
-            // we should be searching on.
-            return Err(Error::CursorOnAmbiguousPair);
+            // Cursor is directly on match char and we don't have any tree-sitter syntax available
+            // to help identify the matching pair. We return no match because there's no way to know
+            // which side of the char we should be searching on.
+            let syntax = syntax.ok_or(Error::CursorOnAmbiguousPair)?;
+
+            let match_pos = find_matching_bracket(syntax, &text.into(), pos)
```

Comment:
This line is the source of the current merge conflict. The RopeSlice can be passed on through now.

```suggestion
            let match_pos = find_matching_bracket(syntax, text, pos)
```